core/src/templates/components/media/media.twig

```diff
@@ -0,0 +1,80 @@
+{#
+/**
+ * @file
+ * Media Component
+ *
+ * A media element, e.g., image, video, audio, iframe etc. The default .su-media component takes custom HTML for the media content.
+ *
+ * Available variants:
+ * .su-media--image - Image
+ * .su-media--video - HTML5 video
+ * .su-media--audio - HTML5 audio
+ *
+ * Available variables:
+ * - attributes: Additional HTML attributes not already provided.
+ * - modifier_class: Additional CSS classes to change look and behavior.
+ * - media_wrapper_modifier_class: Additional CSS classes to change look and behavior of the media wrapper, e.g. "su-aspect-ratio" to provide aspect ratio constraint.
+ * - media_link: Optional URL for the media element to link to (only works for images or <video> with no controls).
+ * - media_image_src: The src link to the image file.
+ * - media_image_srcset: The srcset of the image if exists.
+ * - media_video_src: The src link to the video file.
+ * - media_video_poster: Optional preview image of the video.
+ * - media_audio_src: The src link to the audio file.
+ * - media_type: Type of the media file, e.g. mp4, webm, wav (self-hosted video or audio only).
+ * - media_custom: An open field for the full HTML element of the media content, e.g., an embedded YouTube iframe.
+ * - media_content_attributes: Additional HTML attributes for the media content, e.g., for <video> - controls, autoplay; for <img> - sizes (when srcset is provided), alt, width, height.
+ * - media_caption: Optional caption for the media.
+ */
+#}
+<figure {{ attributes }} class="su-media {{ modifier_class }}">
+
+  {# Option to make the media element a clickable link #}
+  {%- if media_link -%}
+    <a href="{{ media_link }}">
+  {%- endif -%}
+
+  {# Wrapper for the media element where aspect ratio mixin can be applied to #}
+  <div class="su-media__wrapper {{ media_wrapper_modifier_class }}">
+
+    {%- if 'su-media--video' in modifier_class %}
+      {# Find the video type from the src file extension #}
+      {% set media_type = media_video_src | split('.') | last %}
```

Comment:
```suggestion
      {% set media_type = media_video_src|split('.')|last %}
```
No spaces in chained filters according to the coding style guide: https://twig.symfony.com/doc/2.x/coding_standards.html

Reply:
Also, some src values may be URLs without an extension and this method would not be able to derive the mimetype you are looking for. I would change this variable to first, allow for a value to be passed in to the template, and if not provided, then look for an extension, and if those two options are empty, default to something sensible.